asio/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.14...3.23)
+
+include(cmake/prelude.cmake)
+
+project(asio
+        VERSION 1.22.1
+        DESCRIPTION "Asio C++ Library"
+        HOMEPAGE_URL "http://think-async.com/Asio/"
+        LANGUAGES CXX
+)
+
+include(cmake/project-is-top-level.cmake)
+include(cmake/variables.cmake)
+
+if(PROJECT_IS_TOP_LEVEL)
+  include(GNUInstallDirs) # for CMAKE_INSTALL_INCLUDEDIR
+endif()
+
+# ---- Declare library ----
+
+add_library(asio INTERFACE)
+add_library(asio::asio ALIAS asio)
+target_include_directories(
+  asio SYSTEM # XXX ${warning_guard}
+  INTERFACE "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>" #
+            "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
```

Comment:
I missed this `SYSTEM` include!

Reply:
`INSTALL_INTERFACE` is unnecessary here. The `install(TARGETS)`'s special form (`INCLUDE DESTINATION`) already adds the include path to the install interface.

`SYSTEM` is already added with the `warning_guard` variable when necessary. `warning_guard` is for vendoring projects, see the comment in `variables.cmake`.